pandas/tests/window/test_rolling.py

```diff
@@ -465,3 +466,35 @@ def test_rolling_count_default_min_periods_with_null_values(constructor):
     result = constructor(values).rolling(3).count()
     expected = constructor(expected_counts)
     tm.assert_equal(result, expected)
+
+
+@pytest.mark.parametrize("constructor", [Series, DataFrame])
+@pytest.mark.parametrize(
+    "func,expected",
+    [
+        ("min", [0.0, 1.0, 2.0, 3.0, 4.0, 5.0, 6.0, 7.0, 8.0, np.nan]),
+        ("max", [2.0, 3.0, 4.0, 5.0, 6.0, 7.0, 8.0, 9.0, 9.0, np.nan]),
+    ],
+)
+def test_rolling_forward_window_min(constructor, func, expected):
+    # GH 32865
+    class ForwardIndexer(BaseIndexer):
+        def get_window_bounds(self, num_values, min_periods, center, closed):
+            start = np.empty(num_values, dtype=np.int64)
+            end = np.empty(num_values, dtype=np.int64)
+            for i in range(num_values):
+                if i + min_periods <= num_values:
+                    start[i] = i
+                    end[i] = min(i + self.window_size, num_values)
+                else:
+                    start[i] = i
+                    end[i] = i + 1
+            return start, end
+
+    values = np.arange(10)
+
+    indexer = ForwardIndexer(window_size=3)
```

Comment:
hmm are you anticpating exposing this to users? should we be using this internally?

Reply:
Do you mean exposing a forward-looking rolling window as part of our default API?
I don't think that is necessary: my goal is to make sure that our implementations are robust against custom indexers with arbitrary offsets relative to the position in the input array they are calculated for.
The `ForwardIndexer` class is used in testing for convenience as it is, by far, the most common customization people expect us to support, in my limited experience.

Reply:
Might be interesting to provide a standard set of `BaseIndexer`s in the library. `ForwardIndexer` may be one that we provide once we standardize the definition and API.

@AlexKirko feel free to open a new issue about including `ForwardIndexer` and `BackwardIndexer` in the `pandas.api.indexers`

Reply:
Thanks, created a new issue. Will be happy to submit a PR in a day or two.